src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -76,6 +76,20 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor, ILoggerFactory
         /// </summary>
         public int Count => _coherentState.Count;
 
+        /// <summary>
+        /// Gets an enumerable of the all the keys in the <see cref="MemoryCache"/>.
+        /// </summary>
+        public IEnumerable<object> Keys
```

Comment:
Did you consider returning `ICacheEntry` rather than just `Keys`? Is my understanding correct that some people want to make decisions based on `ICacheEntry.Value` https://github.com/dotnet/extensions/issues/4766#issuecomment-1845802796?

Reply:
This can also be achieved by implementing `IEnumerable<KeyValuePair<object, ICacheEntry>>` as you mentioned in the proposal, but doing it in the property doesn't clutter the IDE with LINQ.

Reply:
I did (https://github.com/dotnet/runtime/issues/36554#issuecomment-1818437941):

> that would allow to mutate cache entries (values) after adding them to the cache, which is not supported and can lead into issues like #88733.

Reply:
Good point, I didn't notice ICacheEntry had writable properties.